web/blueprints/user/__init__.py

```diff
@@ -142,15 +145,119 @@ def static_datasheet(user_id):
 
 @bp.route('/json/search')
 def json_search():
-    query = request.args['query']
-    results = session.session.query(User.id, User.login, User.name).filter(or_(
-        func.lower(User.name).like(func.lower(u"%{0}%".format(query))),
-        func.lower(User.login).like(func.lower(u"%{0}%".format(query))),
-        cast(User.id, Text).like(u"{0}%".format(query))
-    )).all()
-    users = [{"id": user_id, "login": login, "name": name}
-             for user_id, login, name in results]
-    return jsonify(users=users)
+    user_id = request.args.get('id')
+    name = request.args.get('name')
+    login = request.args.get('login')
+    mac = request.args.get('mac')
+    ip_address = request.args.get('ip_address')
+    trafficgroup_id = request.args.get('trafficgroup_id')
+    propertygroup_id = request.args.get('propertygroup_id')
+    building_id = request.args.get('building_id')
+    query = request.args.get("query")
+    result = User.q
+
+    applied_filter = True
+
+    if user_id and represents_int(user_id):
```

Comment:
This still requires a user_id != "" check. There would be always thrown a 400 error otherwise, even if user_id was empty.

Reply:
Thinking about it, it would be best, to use `WTForms` here, as you are handling form input. There is no need to reimplement input validation. You should need to set the `{ method: 'post' }` option for bootstrap-table and define a `Form`.

Reply:
You can just then validate the input once with `validate` and raise the `BadRequest` error and return a detailed error message with all errors from `form.errors`.